### PR TITLE
chore(application-system): Payments updates to lifecycle on callback, UX fixes.

### PR DESCRIPTION
**Pull request by obmagnusson**
## What

- Extending the lifecycle for payment states on payment callback success. 
- Added loading dots when waiting on payment callback
- UI updates when applicaiton fails to transition from payment state.

## Why

Better UX. 
Problem: Application users may experience disruptions after a successful payment due to a service failure. Currently, the payment state lifecycle is limited to 24 hours, which may not be enough time to resolve the issue.

Solution: We will extend the payment state lifecycle, allowing users to continue with the application once the fault is resolved.

## Screenshots / Gifs
image
image

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Introduced a new tag for payment callback endpoints in the API documentation for better organization.
	- Enhanced the `Callback` structure with validation and documentation annotations.
	- Improved payment approval logic to manage application data retention based on payment status.
	- Added a new error message for payment submission failures to enhance user feedback.

- **Bug Fixes**
	- Improved error handling in the payment processing component for clearer user notifications.

### CODE REVIEW

File: libs/application/api/payment/src/lib/payment-callback.controller.ts [hunk 1]
```diff
@@ -21,5 +27,18 @@ export class PaymentCallbackController {
       callback.receptionID,
       applicationId,
     )
+
+    const application = await this.applicationService.findOneById(applicationId)
+    if (application) {
+      const oneMonthFromNow = new Date()
+      oneMonthFromNow.setMonth(oneMonthFromNow.getMonth() + 1)
```
Comment (Toti91) [inline libs/application/api/payment/src/lib/payment-callback.controller.ts:34]: What if current month is 12, wont this be setting the month to 13?

I would just use `date-fns`

```ts
import addMonths from 'date-fns/addMonths'

const oneMonthFromNow = addMonths(new Date(), 1)
```

File: libs/application/ui-components/src/components/PaymentPending/PaymentPending.tsx [hunk 2]
```diff
@@ -84,19 +89,45 @@ export const PaymentPending: FC<
...
+          justifyContent="center"
+          style={{
+            height: 400,
+          }}
+        >
+          <AlertMessage
+            type="error"
+            title={msg(coreErrorMessages.paymentSubmitFailed)}
+            message="Villa hefur komið upp við áframhaldandi vinnslu. Vinsamlegast reynið aftur síðar. Ef villa endurtekur sig vinsamlegast hafið samband við island@island.is."
+          />
```
Comment (Toti91) [inline libs/application/ui-components/src/components/PaymentPending/PaymentPending.tsx:106]: add this text to `coreMessages` so that it is translatable

**Review by jonnigs — APPROVED** [review]
LGTM!

**Review by Toti91 — CHANGES_REQUESTED** [review]
TWo things

**Pull request merged**